tap_gong/client.py

```diff
@@ -39,14 +40,39 @@ def response_error_message(self, response: requests.Response) -> str:
         """
             Method overridden to capture errors correctly based on gong API response.
         """
-        full_path = urlparse(response.url).path or self.path
-        if 400 <= response.status_code < 500:
-            error_type = "Client"
+        # Unauthorized for path|Validate credentials failed|Missing mandatory header "Authorization"
+        if response.status_code == 401:
+            return 'The key or secret provided is incorrect.'
+        # No users found corresponding to the provided filters
+        if response.status_code == 404:
+            return 'There was no data found for the given date range.'
+
+        errors = None
+        try:
+            errors = json.loads(response.content.decode('utf-8')).get("errors", [])
+        except:
+            pass
+        
+        if errors is None or len(errors) == 0:
+            msg = 'Unexpected error'
         else:
-            error_type = "Server"
-        errors = json.loads(response.content.decode('utf-8')).get("errors", [])
+            msg = ', '.join(errors)
+        
+        return f'Import failed with following Gong error: {msg}'
```

Comment:
example: 
`singer_sdk.exceptions.FatalAPIError: Import failed with following Gong error: Validate credentials failed. Please check your credentials and try again.`
- generated by commenting out `if response.status_code == 401:` block as I couldn't get any errors other than the ones we handled already.

err doc: no error doc